Include/XAsync.h

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#if !defined(__cplusplus)
+   #error C++11 required
+#endif
+
+#pragma once
+#include <XTaskQueue.h>
+
+extern "C"
+{
+
+/// <summary>
+/// An XAsyncBlock defines a piece of asynchronous work.  An async block can be used
+/// to poll for an async call's status and retrieve call results.  An XAsyncBlock
+/// needs to remain valid for the lifetime of an async call.
+/// </summary>
+struct XAsyncBlock;
+
+/// <summary>
+/// Callback routine that is invoked when an async call completes. Use XAsyncGetStatus
+/// or the async call's Get*Result method to obtain the results of the call.  If the
+/// call was canceled these methods will return E_ABORT.
+/// </summary>
+/// <param name='asyncBlock'>A pointer to the XAsyncBlock that was passed to the async call.</param>
+/// <seealso cref='XAsyncBlock' />
+typedef void CALLBACK XAsyncCompletionRoutine(_Inout_ struct XAsyncBlock* asyncBlock);
+
+/// <summary>
+/// Callback routine that is invoked on a worker asynchronously when XAsyncRun is called.
+/// The result of the callback becomes the result of the async call.
+/// </summary>
+/// <param name='asyncBlock'>A pointer to the XAsyncBlock that was passed to XAsyncRun.</param>
+/// <seealso cref='XAsyncBlock' />
+/// <seealso cref='XAsyncRun' />
+typedef HRESULT CALLBACK XAsyncWork(_Inout_ struct XAsyncBlock* asyncBlock);
+
+struct XAsyncBlock
+{
+    /// <summary>
+    /// The queue to queue the call on
+    /// </summary>
+    XTaskQueueHandle queue;
```

Comment:
is this optional again?

Reply:
Yes, but selectively so. IE, an app can set nullptr into the process task queue and then this parameter is no longer optional.  Basically, internal code uses the process task queue if this member is null, and if it doesn't get a valid one it errors with E_NO_TASK_QUEUE.